### PR TITLE
feat(#2611): add segmented variant to tabs

### PR DESCRIPTION
**Pull request by vanessatran-ddi**
# Before (the change)

 - Tabs component only supported the default underline-style variant
  - Individual tabs could not be disabled - all tabs were always selectable
  - No support for skipping tabs during keyboard navigation
  - No `variant` prop on `<GoabTabs>`
  - No `disabled` prop on `<GoabTab>`
  
# After (the change)
- Added `variant` prop to `<GoabTabs>` with two options:
    - `"default"` - Standard underline-style tabs (existing behavior)
    - `"segmented"` - New pill-style tabs with animated sliding background indicator
  - Added `disabled` prop to `<GoabTab>`:
    - Disabled tabs cannot be clicked or selected
    - Disabled tabs are skipped during keyboard navigation (Arrow keys, Home, End)
    - Disabled tabs display with grayed-out styling and `cursor: not-allowed`
    - If `initialTab` points to a disabled tab, the first enabled tab is automatically selected
  - Added accessibility attributes for disabled tabs (`aria-disabled="true"`, `tabindex="-1"`)
  - Segmented variant uses MutationObserver and ResizeObserver to resize the pill when tab content changes (e.g., badge count updates)
  
image

## Make sure that you've checked the boxes below before you submit the PR

- [x] I have read and followed the [setup steps](https://goa-dio.atlassian.net/wiki/spaces/DS/pages/3086385489/Contribution+-+web+content#%5BinlineExtension%5DCode-contributions)
- [x] I have created necessary unit tests
- [x] I have tested the functionality in both React and Angular.

### React
  1. Run `npx nx serve prs-react`
  2. Navigate to `/features/2611` to test:
     - Segmented variant vs default variant comparison
     - Long labels handling
     - Disabled tabs in both variants
     - Edge case: initialTab pointing to disabled tab
  3. Navigate to `/features/2611-tabs-disabled` to verify:
     - When `initialTab=1` and Tab 1 is disabled, Tab 2 is automatically selected

  ### Angular
  1. Run `npx nx serve prs-angular`
  2. Navigate to `/features/2611` to test the same scenarios as React
  3. Navigate to `/features/2611-tabs-disabled` to verify the disabled tab edge case

### CODE REVIEW

**Review by willcodeforcoffee — DISMISSED** [review]
Looks good! Great demos, too! Really impressive. 😃

File: libs/web-components/src/components/tabs/Tabs.svelte [hunk 1]
```diff
@@ -152,7 +220,9 @@
   function handleHashChange() {
     const tabIndexFromHash = getTabIndexFromHash();
```
Comment (ArakTaiRoth) [inline libs/web-components/src/components/tabs/Tabs.svelte]: So, the hash can be changed to a disabled tab's hash, this doesn't activate the disabled tab (good), but it does change the URL to include the hash of the disabled tab (bad)

This can be seen in the Test Case 3 scenario I added to the `tabs-disabled` pr files.

Do we want to have it revert the hash shown in the URL to the last good activated hash? @chrisolsen

File: libs/web-components/src/components/tabs/Tabs.svelte [hunk 2]
```diff
@@ -230,6 +321,33 @@
...
+    const tabs = _tabsEl.querySelectorAll('[role="tab"]');
+    const selectedTab = tabs[_currentTab - 1] as HTMLElement;
+
+    if (selectedTab) {
+      const tabsRect = _tabsEl.getBoundingClientRect();
+      const selectedRect = selectedTab.getBoundingClientRect();
+
+      _segmentedIndicatorLeft = selectedRect.left - tabsRect.left - 1;
+      _segmentedIndicatorWidth = selectedRect.width;
+      _segmentedIndicatorHeight = selectedRect.height;
```
[the remark this answers is not in the document]
Comment (vanessatran-ddi) [reply libs/web-components/src/components/tabs/Tabs.svelte]: @chrisolsen 
the sliding animation needs js because the indicator has to know where to move and how is the width. Each tab has different width depending on text length and badge (or other react node), if I use only css, it won't have the animation.


https://github.com/user-attachments/assets/eae91d35-cff2-4424-8f27-df8beb3a12e1

**Comment by vanessatran-ddi** [comment]
> @vanessatran-ddi Looks good overall, great work.
> 
> The only additional thing I noticed:
> 
> * on the v1 version, the colour of the text for both the active tab and the inactive tab is too light. Lets adjust it to greyscale-800 on the inactive tab, and text-default on the active tab.
> 
> ### Current:
> Image
> ### Update to:
> Image

Hi @twjeffery  I addressed your comment on [this commit ](https://github.com/GovAlta/ui-components/pull/3302/commits/59079b4f8cdd7c8a5ec0466a368086e78dd3ace9)

**Review by twjeffery — APPROVED** [review]
